### PR TITLE
feat(nuxi): auto cleanup with project manifest changes

### LINKED ISSUE #6668: clean up up cache dirs on `rootDir`/hash change
we've had a number of issues tied back somehow to cache not being properly invalidated, such as when a project directory is moved or renamed, a transform is updated within nuxt, or a downstream package is updated. These seem to tie back to invalid caches.

Vite already invalidates its cache in certain conditions:
https://github.com/vitejs/vite/blob/ed0286186b24748ec7bfa336f83c382363a22f1d/packages/vite/src/node/optimizer/index.ts#L1034-L1055

This issue is to track the possible causes of cache failure and invalidate them ourselves.

Possibly related: https://github.com/nuxt/framework/issues/6623

---

Current solution: Run `npx nuxi-edge@latest cleanup` after dependency upgrades and before `nuxt dev` manually to ensure caches are cleaned up.

### PR DESCRIPTION
**Pull request by pi0**
### 🔗 Linked issue

Resolves #6668 and adds basic manifest implementing #6670

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

When a major change in a project happens (such as updating nuxt or renaming directory) we most likely need to invalidate caches to avoid unpredictable behavior. This PR does two things:

- Write a basic `nuxt.json` to `buildDir`/`.nuxt` only for `nuxi dev` (1)
- Checks manifest hash (full hash) against the previous build and call `nuxi cleanup` in case of hash mismatches.

Remarks:
- I've not included more extensive manifest info and hashes unless we find initial change is not enough. As mentioned by @danielroe in #6668, bundlers (vite and webpack) also handle cache invalidation for dependencies
- We probably need to generate two separate manifests for dev and prod (https://github.com/nuxt/framework/issues/6671)
- It only invalidates caches for `nuxi dev` and not `nuxi build` because i don't believe we saw any cache issue on `nuxi build`. We can add later on to both commands and utils are already refactored to prepare for this.

(1) Sample `.nuxt/nuxt.json`:

```json
{
  "_hash": "AQBCqiYugT",
  "project": {
    "rootDir": "/Users/pooya/Code/framework/playground"
  },
  "versions": {
    "nuxt": "3.0.0-rc.8"
  }
}
```

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxi/src/commands/dev.ts [hunk 1]
```diff
@@ -75,6 +76,13 @@ export default defineNuxtCommand({
           showURL()
         }
 
+        // Check if we need cache invalidation
+        const previousManifest = await loadNuxtManifest(currentNuxt.options.buildDir)
+        const newManifest = await writeNuxtManifest(currentNuxt)
+        if (previousManifest && newManifest && previousManifest._hash !== newManifest._hash) {
+          await cleanupNuxtDirs(currentNuxt.options.rootDir)
+        }
```
Comment (danielroe) [inline packages/nuxi/src/commands/dev.ts:84]: Do you think we need to do this within the refresh loop? I imagine you are thinking if project dependencies are updated while dev server is running? Or might it be enough to do it at start of dev command?

Comment (danielroe) [reply packages/nuxi/src/commands/dev.ts:84]: If you are imagining runtime updating deps, then is there any merit in watching `package.json`? I don't think we can directly add to this line as it's relative to `srcDir`.

https://github.com/nuxt/framework/blob/75e3be9d0eefabf6d65463c6457349670bb32e02/packages/nuxi/src/commands/dev.ts#L105

File: packages/nuxi/src/utils/nuxt.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  return manifest
+}
+
+export async function loadNuxtManifest (buildDir: string): Promise<NuxtProjectManifest | null> {
+  const manifestPath = resolve(buildDir, 'nuxt.json')
+  const manifest: NuxtProjectManifest | null = await fsp.readFile(manifestPath, 'utf-8')
+    .then(data => JSON.parse(data) as NuxtProjectManifest)
+    .catch(() => null)
+  return manifest
+}
```
Comment (danielroe) [inline packages/nuxi/src/utils/nuxt.ts:58]: Maybe make it more explicit?

```suggestion
export async function writeNuxtManifest (nuxt: Nuxt): Promise<NuxtProjectManifest> {
  const manifest = resolveNuxtManifest(nuxt)
  const manifestPath = resolve(nuxt.options.buildDir, 'nuxt.manifest.json')
  await fsp.mkdir(dirname(manifestPath), { recursive: true })
  await fsp.writeFile(manifestPath, JSON.stringify(manifest, null, 2), 'utf-8')
  return manifest
}

export async function loadNuxtManifest (buildDir: string): Promise<NuxtProjectManifest | null> {
  const manifestPath = resolve(buildDir, 'nuxt.manifest.json')
  const manifest: NuxtProjectManifest | null = await fsp.readFile(manifestPath, 'utf-8')
    .then(data => JSON.parse(data) as NuxtProjectManifest)
    .catch(() => null)
  return manifest
}
```

File: packages/nuxi/src/commands/dev.ts (hunk 1 above)
Comment (pi0) [reply packages/nuxi/src/commands/dev.ts:84]: Indeed at the moment doesn't benefits reloads. Will make it once.

File: packages/nuxi/src/utils/nuxt.ts (hunk 2 above)
Comment (pi0) [reply packages/nuxi/src/utils/nuxt.ts:58]: Nitro uses the same `nitro.json` format and we also generate `dist/module.json` for module-builder output. "Manifest" itself is an unexplicit name IMO to assign (ie: what is exactly manifest).

File: packages/nuxi/src/commands/dev.ts (hunk 1 above)
Comment (pi0) [reply packages/nuxi/src/commands/dev.ts]: b1e86568897748f17d462faa76f02673c96b0c64

**Pull request merged**